lib/rspec/core.rb

```diff
@@ -9,6 +9,46 @@
 
 RSpec::Support.define_optimized_require_for_rspec(:core) { |f| require_relative f }
 
+module RSpec
+  module Core
+    # @private
+    class LookupSet
+      include Enumerable
+
+      def initialize(array=[])
+        @values = {}
+        merge(array)
+      end
+
+      def <<(key)
+        @values[key] = true
+      end
+
+      def each(&block)
+        @values.keys.each(&block)
+      end
+
+      def to_a
+        @values.keys
+      end
+
+      def +(others)
+        LookupSet.new(@values).merge(others)
+      end
+
+      def include?(key)
+        @values[key] == true
```

Comment:
This can just be `@values.key?(key)` which might be faster (and also seems simpler).